### ISSUE TITLE
[FEAT]: Russian localization

### ISSUE THREAD

**Issue by younyokel** [BartoszCichecki/LenovoLegionToolkit#109]
Now that the Lenovo Legion Toolkit has been redesigned, are there plans to support multiple languages?

**Comment by BartoszCichecki** [comment]
Sure, why not. Which ones are you thinking about?

**Comment by younyokel** [comment]
I'd like to see Russian.

**Comment by BartoszCichecki** [comment]
I don't speak Russian, so I will need some help 😉 I will keep this issue open and will.mentiom you once I have the list of strings to translate.

**Comment by Cryoshell99** [comment]
Hi, I made a [fork](https://github.com/Cryoshell99/LenovoLegionToolkitRU) of your project and manually changed the interface elements to Russian. I haven't compiled the resulting code yet. Please see what settings have been changed (I marked the commits as RU), I hope this will help with further translation into other languages.

**Comment by BartoszCichecki** [comment]
Due to amount of work required to move all strings into resources, and ongoing cost of maintaining translations with new features I decided to put this one on hold.

Thanks @Cryoshell99 for your work, I will definitely use it once I pick it up.

**Comment by MZhao-ouo** [comment]
@BartoszCichecki Thank you for your work!

 I am trying to add Language Change function by using dynamic resource. But I met some trouble: it doesn't work in `Title` but works well in `Text`. I had googled a lot but still no progress. Could you give me some suggestion or direction?

![image](https://user-images.githubusercontent.com/70903329/192579264-8a3a6173-117b-44d9-9042-a1df5774bcc1.png)

**Comment by BartoszCichecki** [comment]
If you could wait a little bit with that would be great. There is a guy already doing a Russian translation. And if you both do it there will be plenty of conflicts.

**Comment by MZhao-ouo** [comment]
I would like to do a Chinese translation and add a language switcher. Is someone doing the same?

**Comment by BartoszCichecki** [comment]
Yeah, there is one guy who is extracting the strings and preparing the switcher (plus RU translation), which needs to be done first, once this is done the code will be ready to add more languages :)

**Comment by BartoszCichecki** [comment]
Of course, once code is prepared for more languages I will ping you!

**Comment by BeYkeRYkt** [comment]
That's the guy right there. Anyway, it's still a work in progress. At this stage, we have the ability to change the language from the enum list in runtime, but not all text has been moved to resources yet and there are some problems with the interface elements.

https://user-images.githubusercontent.com/2690323/192832322-1c32cc6e-af67-4bc7-abc3-cdd6b2397899.mp4

I will try to upload the source code in the near future after cleaning it up.

**Comment by MZhao-ouo** [comment]
Waiting for your good news and I can make a contribution.

**Comment by BartoszCichecki** [comment]
@BeYkeRYkt how is this progressing? :)

**Comment by younyokel** [comment]
> @BeYkeRYkt how is this progressing? :)

Not to be rude or interrupt @BeYkeRYkt's work, but can you still create a Russian translation in Crowdin?

**Comment by BartoszCichecki** [comment]
Translations can be done on Crowdin. You can check the readme how to join the project and contribute.

**Comment by BartoszCichecki** [comment]
Since translation is in progress on Crowdin, I am closing this issue.
